# ELLY OKUMU OTHIM 

# WEEK 8

# INDIVIDUAL PROJECT 

In [1]:
#START BY IMPORTING PANDAS AND NUMPY FOR ARRAYS 

import pandas as pd
import numpy as np

In [2]:
#CREAT A DATAFRAME FROM THE DICTIONARY GIVEN
dict1 = [{'product_id':23, 'name':'computer', 'wholesale_price': 500, 'retail_price':1000, 'sales':100}, 
{'product_id':96, 'name':'Python Workout', 'wholesale_price': 35,'retail_price':75, 'sales':1000},
{'product_id':97, 'name':'Pandas Workout', 'wholesale_price': 35, 'retail_price':75, 'sales':500},
{'product_id':15, 'name':'banana', 'wholesale_price': 0.5,'retail_price':1, 'sales':200},
{'product_id':87, 'name':'sandwich', 'wholesale_price': 3,'retail_price':5, 'sales':300}]

df = pd.DataFrame(dict1)

#print(dict1)
print(df)

   product_id            name  wholesale_price  retail_price  sales
0          23        computer            500.0          1000    100
1          96  Python Workout             35.0            75   1000
2          97  Pandas Workout             35.0            75    500
3          15          banana              0.5             1    200
4          87        sandwich              3.0             5    300


In [3]:
#Run the dataframe created , just to check how it looks 
df

product_id            name  wholesale_price  retail_price  sales
0          23        computer            500.0          1000    100
1          96  Python Workout             35.0            75   1000
2          97  Pandas Workout             35.0            75    500
3          15          banana              0.5             1    200
4          87        sandwich              3.0             5    300

## QUESTION 1
### Calculate the Total Profit for each product 

In [4]:
# Required to find the total product for individual products
#net_revenue_per_product = (retail_price - wholesale price) * sales


df['net_revenue_per_product'] = (df['retail_price'] - df['wholesale_price']) * df['sales']

In [5]:
#Show the new dataframe created with the new column of net_revenue_per_product
df

product_id            name  wholesale_price  retail_price  sales  \
0          23        computer            500.0          1000    100   
1          96  Python Workout             35.0            75   1000   
2          97  Pandas Workout             35.0            75    500   
3          15          banana              0.5             1    200   
4          87        sandwich              3.0             5    300   

   net_revenue_per_product  
0                  50000.0  
1                  40000.0  
2                  20000.0  
3                    100.0  
4                    600.0

## QUESTION 2
#### How much total net revenue you received from all of these sales?

In [6]:
# We're supposed to find out the total revenue from all the sales i.e sum the revenue_per_product column

print ( 'The total revenue of all the sales is : ', df.net_revenue_per_product.sum() ,'$')

The total revenue of all the sales is :  110700.0 $


## QUESTION 3 
### What product is product retail price more than twice the wholesale price?

In [7]:
# finding the value in the relail price more than twice in the whole sale price, index to its product id or name 

product1 = df[df['retail_price'] > 2*df['wholesale_price']
]

In [8]:
# print out the results of the answer asked 

#Pass in a df to make sure the value returned is not a boolean value 

product1

product_id            name  wholesale_price  retail_price  sales  \
1          96  Python Workout             35.0            75   1000   
2          97  Pandas Workout             35.0            75    500   

   net_revenue_per_product  
1                  40000.0  
2                  20000.0

In [9]:
#printing the whole dataframe , to view easily for the next question 
df

product_id            name  wholesale_price  retail_price  sales  \
0          23        computer            500.0          1000    100   
1          96  Python Workout             35.0            75   1000   
2          97  Pandas Workout             35.0            75    500   
3          15          banana              0.5             1    200   
4          87        sandwich              3.0             5    300   

   net_revenue_per_product  
0                  50000.0  
1                  40000.0  
2                  20000.0  
3                    100.0  
4                    600.0

## QUESTION 4 
## How much did the store make from food vs. computers vs. books?

In [10]:
# Grouping the company's profit by food, computer and books 
# METHOD 1


computer = ['computer']
food = ['banana', 'sandwich' ]
book = ['Python Workout' , 'Pandas Workout']



In [11]:
#USING THE ISIN FUNCTION TO FIND THE SPECIFIC 

computer_sum = df[df.name.isin(computer)].net_revenue_per_product.sum()
food_sum = df[df.name.isin(food)].net_revenue_per_product.sum()
book_sum = df[df.name.isin(book)].net_revenue_per_product.sum()

In [12]:
#Run the cells just to have a look 

print('computer_sum', computer_sum)
print('food_sum', food_sum)
print('food_sum' , book_sum)

computer_sum 50000.0
food_sum 700.0
food_sum 60000.0


In [13]:
#create a dictionary , pass in values the call that dictionary 

DICT = {"Computer" : computer_sum , "Food  " : food_sum, "books  " : book_sum}

#print(DICT)


In [14]:
#CONVERT IT TO A DF
KIM = pd.Series(DICT)

KIM

Computer    50000.0
Food          700.0
books       60000.0
dtype: float64

## QUESTION 5 
### Because your store is doing so well, you’re able to negotiate a 30% discount on the wholesale price of goods. Calculate the new net revenue

In [15]:
#Step 1 - Display your DF 
#Step 2 - Calculate the new price of the wholesale price and add it as a new column 
#Step 3 - Calculate the new  net revenue based on the new discounted wholesale price 

df 

product_id            name  wholesale_price  retail_price  sales  \
0          23        computer            500.0          1000    100   
1          96  Python Workout             35.0            75   1000   
2          97  Pandas Workout             35.0            75    500   
3          15          banana              0.5             1    200   
4          87        sandwich              3.0             5    300   

   net_revenue_per_product  
0                  50000.0  
1                  40000.0  
2                  20000.0  
3                    100.0  
4                    600.0

In [16]:
#Step 2 - Calculate the new price of the wholesale price and add it as a new column 
df['wholesale@30%discount'] = df['wholesale_price']*0.7

df

product_id            name  wholesale_price  retail_price  sales  \
0          23        computer            500.0          1000    100   
1          96  Python Workout             35.0            75   1000   
2          97  Pandas Workout             35.0            75    500   
3          15          banana              0.5             1    200   
4          87        sandwich              3.0             5    300   

   net_revenue_per_product  wholesale@30%discount  
0                  50000.0                 350.00  
1                  40000.0                  24.50  
2                  20000.0                  24.50  
3                    100.0                   0.35  
4                    600.0                   2.10

In [17]:
#Step 3 - Calculate the new  net revenue based on the new discounted wholesale price 
#i.e the sum of the new column 

df['new_revenue_per_product'] = (df['retail_price']- df['wholesale@30%discount']) * df['sales']

df

product_id            name  wholesale_price  retail_price  sales  \
0          23        computer            500.0          1000    100   
1          96  Python Workout             35.0            75   1000   
2          97  Pandas Workout             35.0            75    500   
3          15          banana              0.5             1    200   
4          87        sandwich              3.0             5    300   

   net_revenue_per_product  wholesale@30%discount  new_revenue_per_product  
0                  50000.0                 350.00                  65000.0  
1                  40000.0                  24.50                  50500.0  
2                  20000.0                  24.50                  25250.0  
3                    100.0                   0.35                    130.0  
4                    600.0                   2.10                    870.0

## Running the revenue before and after revenue to confirm if everything works 

In [18]:
#Total Revenue 
print('The new net revenue before discount from wholesaler $' ,df.new_revenue_per_product.sum())
print('The net net revenue before discount from wholesaler $' ,df.net_revenue_per_product.sum())

The new net revenue before discount from wholesaler $ 141750.0
The net net revenue before discount from wholesaler $ 110700.0


# Continuation to individual project ............ 

# INDIVIDUAL PROJECT 

#### Our local government is thinking about imposing a sales tax, and is thinking about 15, 20, and 25 percent rates. Show how much

#### less you would net with each of these tax amounts by adding columns to the data frame for current income, as well.

In [19]:
df

product_id            name  wholesale_price  retail_price  sales  \
0          23        computer            500.0          1000    100   
1          96  Python Workout             35.0            75   1000   
2          97  Pandas Workout             35.0            75    500   
3          15          banana              0.5             1    200   
4          87        sandwich              3.0             5    300   

   net_revenue_per_product  wholesale@30%discount  new_revenue_per_product  
0                  50000.0                 350.00                  65000.0  
1                  40000.0                  24.50                  50500.0  
2                  20000.0                  24.50                  25250.0  
3                    100.0                   0.35                    130.0  
4                    600.0                   2.10                    870.0

# WORKING BEFORE DISCOUNT
## 15% TAX 

In [20]:
# FIND THE AMOUNT OF TAX 15% OF REVENUE PER PRODUCT, ADD A NEW COLUMN 
# step 1 - find the amount of tax imposed 
# step 2 - Subtract the net_revenue_per_product - 15%_tax_amount
# step 3 - add the column 
# step 4 - get the difference before and after tax 

df['15%_tax_amount'] = df['net_revenue_per_product'] * 0.15

df 

product_id            name  wholesale_price  retail_price  sales  \
0          23        computer            500.0          1000    100   
1          96  Python Workout             35.0            75   1000   
2          97  Pandas Workout             35.0            75    500   
3          15          banana              0.5             1    200   
4          87        sandwich              3.0             5    300   

   net_revenue_per_product  wholesale@30%discount  new_revenue_per_product  \
0                  50000.0                 350.00                  65000.0   
1                  40000.0                  24.50                  50500.0   
2                  20000.0                  24.50                  25250.0   
3                    100.0                   0.35                    130.0   
4                    600.0                   2.10                    870.0   

   15%_tax_amount  
0          7500.0  
1          6000.0  
2          3000.0  
3            15.0  
4            90.0

In [21]:
#step 2 - Subtract the net_revenue_per_product - 15%_tax_amount

df['revenue_after_15_tax'] = df['net_revenue_per_product'] - df['15%_tax_amount']

df 

product_id            name  wholesale_price  retail_price  sales  \
0          23        computer            500.0          1000    100   
1          96  Python Workout             35.0            75   1000   
2          97  Pandas Workout             35.0            75    500   
3          15          banana              0.5             1    200   
4          87        sandwich              3.0             5    300   

   net_revenue_per_product  wholesale@30%discount  new_revenue_per_product  \
0                  50000.0                 350.00                  65000.0   
1                  40000.0                  24.50                  50500.0   
2                  20000.0                  24.50                  25250.0   
3                    100.0                   0.35                    130.0   
4                    600.0                   2.10                    870.0   

   15%_tax_amount  revenue_after_15_tax  
0          7500.0               42500.0  
1          6000.0               34000.0  
2          3000.0               17000.0  
3            15.0                  85.0  
4            90.0                 510.0

In [22]:
#add the different columns then get the difference 

df.revenue_after_15_tax.sum()

df.net_revenue_per_product.sum()

print('Revenue before 15% tax, without Discount is ', df.revenue_after_15_tax.sum() , '$ ' ,' Revenue after 15% tax, without Discount is ' , df.net_revenue_per_product.sum(), '$' )

Revenue before 15% tax, without Discount is  94095.0 $   Revenue after 15% tax, without Discount is  110700.0 $


In [23]:
difference1 =  df.net_revenue_per_product.sum() - df.revenue_after_15_tax.sum() 

In [24]:
Answer1 = ('Revenue before 15% tax, without Discount is  94095.0 $   Revenue after 15% tax, without Discount is  110700.0 $ with a difference of ', difference1)

In [25]:
print(Answer1)

('Revenue before 15% tax, without Discount is  94095.0 $   Revenue after 15% tax, without Discount is  110700.0 $ with a difference of ', 16605.0)


# 20% WITHOUT DISCOUNT 

In [61]:
# FIND THE AMOUNT OF TAX 20% OF REVENUE PER PRODUCT, ADD A NEW COLUMN 
# step 1 - find the amount of tax imposed 
# step 2 - Subtract the net_revenue_per_product - 20%_tax_amount
# step 3 - add the column 
# step 4 - get the difference before and after tax 

df['20%_tax_amount'] = df['net_revenue_per_product'] * 0.2

df 

product_id            name  wholesale_price  retail_price  sales  \
0          23        computer            500.0          1000    100   
1          96  Python Workout             35.0            75   1000   
2          97  Pandas Workout             35.0            75    500   
3          15          banana              0.5             1    200   
4          87        sandwich              3.0             5    300   

   net_revenue_per_product  wholesale@30%discount  new_revenue_per_product  \
0                  50000.0                 350.00                  65000.0   
1                  40000.0                  24.50                  50500.0   
2                  20000.0                  24.50                  25250.0   
3                    100.0                   0.35                    130.0   
4                    600.0                   2.10                    870.0   

   15%_tax_amount  revenue_after_15_tax  20%_tax_amount  revenue_after_20_tax  \
0          7500.0               42500.0         10000.0               40000.0   
1          6000.0               34000.0          8000.0               32000.0   
2          3000.0               17000.0          4000.0               16000.0   
3            15.0                  85.0            20.0                  80.0   
4            90.0                 510.0           120.0                 480.0   

   25%_tax_amount  revenue_after_25_tax  
0          7500.0               42500.0  
1          6000.0               34000.0  
2          3000.0               17000.0  
3            15.0                  85.0  
4            90.0                 510.0

In [62]:
#step 2 - Subtract the net_revenue_per_product - 20%_tax_amount

df['revenue_after_20_tax'] = df['net_revenue_per_product'] - df['20%_tax_amount']

df 

product_id            name  wholesale_price  retail_price  sales  \
0          23        computer            500.0          1000    100   
1          96  Python Workout             35.0            75   1000   
2          97  Pandas Workout             35.0            75    500   
3          15          banana              0.5             1    200   
4          87        sandwich              3.0             5    300   

   net_revenue_per_product  wholesale@30%discount  new_revenue_per_product  \
0                  50000.0                 350.00                  65000.0   
1                  40000.0                  24.50                  50500.0   
2                  20000.0                  24.50                  25250.0   
3                    100.0                   0.35                    130.0   
4                    600.0                   2.10                    870.0   

   15%_tax_amount  revenue_after_15_tax  20%_tax_amount  revenue_after_20_tax  \
0          7500.0               42500.0         10000.0               40000.0   
1          6000.0               34000.0          8000.0               32000.0   
2          3000.0               17000.0          4000.0               16000.0   
3            15.0                  85.0            20.0                  80.0   
4            90.0                 510.0           120.0                 480.0   

   25%_tax_amount  revenue_after_25_tax  
0          7500.0               42500.0  
1          6000.0               34000.0  
2          3000.0               17000.0  
3            15.0                  85.0  
4            90.0                 510.0

In [70]:
#add the different columns then get the difference 

df.revenue_after_20_tax.sum()

df.net_revenue_per_product.sum()

print('Revenue after 20% tax, without Discount is ', df.revenue_after_20_tax.sum() , '$ ' ,' Revenue before 20% tax, without Discount is ' , df.net_revenue_per_product.sum(), '$' )

Revenue after 20% tax, without Discount is  88560.0 $   Revenue before 20% tax, without Discount is  110700.0 $


In [71]:
difference2 =  df.net_revenue_per_product.sum() - df.revenue_after_20_tax.sum() 

In [72]:
Answer2 = ('Revenue before 20% tax, without Discount is  110700.0 $   Revenue after 20% tax, without Discount is  88560.0 $'
, difference2)

In [73]:
Answer2

('Revenue before 20% tax, without Discount is  110700.0 $   Revenue after 20% tax, without Discount is  88560.0 $',
 22140.0)

In [74]:
difference2

22140.0

### METHOD 2 FOR 20% TAX

In [76]:
#@20%
original_revenue = df.net_revenue_per_product.sum()

original_revenue

110700.0

In [77]:
revenue_20 = df.revenue_after_20_tax.sum()

revenue_20

88560.0

In [78]:
#difference 

original_revenue - revenue_20

22140.0

IN BOTH METHOD 1 AND METHOD 2 WE HAVE ACQUIRED THE SAME RESULTS, OF 22140 AS THE DIFFERENCE THUS CONFIRMING THE METHOD EMPLOYED TO BEING SOUND 

## FOR 20% SALES TAX WITHOUT DISCOUNT, THE STORE NETS 22140 LESS 

# 25% TAX - BEFORE DISCOUNT 

In [37]:
#PRINT OUT THE WHOLE DATASET JUST TO VIEW 

df

product_id            name  wholesale_price  retail_price  sales  \
0          23        computer            500.0          1000    100   
1          96  Python Workout             35.0            75   1000   
2          97  Pandas Workout             35.0            75    500   
3          15          banana              0.5             1    200   
4          87        sandwich              3.0             5    300   

   net_revenue_per_product  wholesale@30%discount  new_revenue_per_product  \
0                  50000.0                 350.00                  65000.0   
1                  40000.0                  24.50                  50500.0   
2                  20000.0                  24.50                  25250.0   
3                    100.0                   0.35                    130.0   
4                    600.0                   2.10                    870.0   

   15%_tax_amount  revenue_after_15_tax  20%_tax_amount  revenue_after_20_tax  
0          7500.0               42500.0          7500.0               42500.0  
1          6000.0               34000.0          6000.0               34000.0  
2          3000.0               17000.0          3000.0               17000.0  
3            15.0                  85.0            15.0                  85.0  
4            90.0                 510.0            90.0                 510.0

In [87]:
# FIND THE AMOUNT OF TAX 25% OF REVENUE PER PRODUCT, ADD A NEW COLUMN 
# step 1 - find the amount of tax imposed 
# step 2 - Subtract the net_revenue_per_product - 25%_tax_amount
# step 3 - add the column 
# step 4 - get the difference before and after tax 

df['25%_tax_amount'] = df['net_revenue_per_product'] * 0.25

df 

product_id            name  wholesale_price  retail_price  sales  \
0          23        computer            500.0          1000    100   
1          96  Python Workout             35.0            75   1000   
2          97  Pandas Workout             35.0            75    500   
3          15          banana              0.5             1    200   
4          87        sandwich              3.0             5    300   

   net_revenue_per_product  wholesale@30%discount  new_revenue_per_product  \
0                  50000.0                 350.00                  65000.0   
1                  40000.0                  24.50                  50500.0   
2                  20000.0                  24.50                  25250.0   
3                    100.0                   0.35                    130.0   
4                    600.0                   2.10                    870.0   

   15%_tax_amount  revenue_after_15_tax  20%_tax_amount  revenue_after_20_tax  \
0          7500.0               42500.0         10000.0               40000.0   
1          6000.0               34000.0          8000.0               32000.0   
2          3000.0               17000.0          4000.0               16000.0   
3            15.0                  85.0            20.0                  80.0   
4            90.0                 510.0           120.0                 480.0   

   25%_tax_amount  revenue_after_25_tax  
0         12500.0               37500.0  
1         10000.0               30000.0  
2          5000.0               15000.0  
3            25.0                  75.0  
4           150.0                 450.0

In [88]:
# step 2 - Subtract the net_revenue_per_product - 25%_tax_amount
#ADD A COLUMN FOR IT, SO THAT YOU CAN ADD THE WHOLE COLUMN

df['revenue_after_25_tax'] = df['net_revenue_per_product'] - df['25%_tax_amount']

df 

product_id            name  wholesale_price  retail_price  sales  \
0          23        computer            500.0          1000    100   
1          96  Python Workout             35.0            75   1000   
2          97  Pandas Workout             35.0            75    500   
3          15          banana              0.5             1    200   
4          87        sandwich              3.0             5    300   

   net_revenue_per_product  wholesale@30%discount  new_revenue_per_product  \
0                  50000.0                 350.00                  65000.0   
1                  40000.0                  24.50                  50500.0   
2                  20000.0                  24.50                  25250.0   
3                    100.0                   0.35                    130.0   
4                    600.0                   2.10                    870.0   

   15%_tax_amount  revenue_after_15_tax  20%_tax_amount  revenue_after_20_tax  \
0          7500.0               42500.0         10000.0               40000.0   
1          6000.0               34000.0          8000.0               32000.0   
2          3000.0               17000.0          4000.0               16000.0   
3            15.0                  85.0            20.0                  80.0   
4            90.0                 510.0           120.0                 480.0   

   25%_tax_amount  revenue_after_25_tax  
0         12500.0               37500.0  
1         10000.0               30000.0  
2          5000.0               15000.0  
3            25.0                  75.0  
4           150.0                 450.0

In [89]:
#add the different columns then get the difference 

df.revenue_after_25_tax.sum()

df.net_revenue_per_product.sum()

print('Revenue after 25% tax, without Discount is ', df.revenue_after_25_tax.sum() , '$ ' ,' Revenue before 25% tax, without Discount is ' , df.net_revenue_per_product.sum(), '$' )

Revenue after 25% tax, without Discount is  83025.0 $   Revenue before 25% tax, without Discount is  110700.0 $


In [90]:
#CALCULATING THE DIFFERENCE

difference3 =  df.net_revenue_per_product.sum() - df.revenue_after_25_tax.sum() 

difference3

27675.0

In [91]:
df

product_id            name  wholesale_price  retail_price  sales  \
0          23        computer            500.0          1000    100   
1          96  Python Workout             35.0            75   1000   
2          97  Pandas Workout             35.0            75    500   
3          15          banana              0.5             1    200   
4          87        sandwich              3.0             5    300   

   net_revenue_per_product  wholesale@30%discount  new_revenue_per_product  \
0                  50000.0                 350.00                  65000.0   
1                  40000.0                  24.50                  50500.0   
2                  20000.0                  24.50                  25250.0   
3                    100.0                   0.35                    130.0   
4                    600.0                   2.10                    870.0   

   15%_tax_amount  revenue_after_15_tax  20%_tax_amount  revenue_after_20_tax  \
0          7500.0               42500.0         10000.0               40000.0   
1          6000.0               34000.0          8000.0               32000.0   
2          3000.0               17000.0          4000.0               16000.0   
3            15.0                  85.0            20.0                  80.0   
4            90.0                 510.0           120.0                 480.0   

   25%_tax_amount  revenue_after_25_tax  
0         12500.0               37500.0  
1         10000.0               30000.0  
2          5000.0               15000.0  
3            25.0                  75.0  
4           150.0                 450.0

#### METHOD 2 FOR 25% TAX

In [92]:
#@20%
original_revenue = df.net_revenue_per_product.sum()

original_revenue

110700.0

In [93]:
#Sum of revenue after 25%
revenue_25 = df.revenue_after_25_tax.sum()

revenue_25

83025.0

In [94]:
#Dfference 
original_revenue - revenue_25

27675.0

IN BOTH METHOD 1 AND METHOD 2 WE HAVE ACQUIRED THE SAME RESULTS, OF 27675 AS THE DIFFERENCE THUS CONFIRMING THE METHOD EMPLOYED TO BEING SOUND 

## FOR 25% SALES TAX WITHOUT DISCOUNT, THE STORE NETS 27675 LESS 

# SUMMATION OF SALES TAX WITHOUT DISCOUNT 

In [97]:
df

product_id            name  wholesale_price  retail_price  sales  \
0          23        computer            500.0          1000    100   
1          96  Python Workout             35.0            75   1000   
2          97  Pandas Workout             35.0            75    500   
3          15          banana              0.5             1    200   
4          87        sandwich              3.0             5    300   

   net_revenue_per_product  wholesale@30%discount  new_revenue_per_product  \
0                  50000.0                 350.00                  65000.0   
1                  40000.0                  24.50                  50500.0   
2                  20000.0                  24.50                  25250.0   
3                    100.0                   0.35                    130.0   
4                    600.0                   2.10                    870.0   

   15%_tax_amount  revenue_after_15_tax  20%_tax_amount  revenue_after_20_tax  \
0          7500.0               42500.0         10000.0               40000.0   
1          6000.0               34000.0          8000.0               32000.0   
2          3000.0               17000.0          4000.0               16000.0   
3            15.0                  85.0            20.0                  80.0   
4            90.0                 510.0           120.0                 480.0   

   25%_tax_amount  revenue_after_25_tax  
0         12500.0               37500.0  
1         10000.0               30000.0  
2          5000.0               15000.0  
3            25.0                  75.0  
4           150.0                 450.0

In [102]:
lost_after_15_discount_per_product = (df.net_revenue_per_product.sum()-df.revenue_after_15_tax.sum())
lost_after_20_discount_per_product = (df.net_revenue_per_product.sum()-df.revenue_after_20_tax.sum())
lost_after_25_discount_per_product = (df.net_revenue_per_product.sum()-df.revenue_after_25_tax.sum())

In [135]:
print('lost after 15 discount per_product KSHS', lost_after_15_discount_per_product)

print('lost after 20 discount per product KSHS', lost_after_20_discount_per_product)

print('lost after 25 discount per product KSHS' , lost_after_25_discount_per_product)

lost after 15 discount per_product KSHS 16605.0
lost after 20 discount per product KSHS 22140.0
lost after 25 discount per product KSHS 27675.0


#### THANK YOU AZUBI FOR THIS OPPORTUNITY, CORRECT WHEREVER I GO WRONG.
#### GOD BLESS YOU MADAM PORTIA, MR. MAWULI 

# WORKING AFTER DISCOUNT IS GIVEN

# WORKING AFTER DISCOUNT IS GIVEN

###  15%  TAX AFTER DISCOUNT  

In [110]:
df

product_id            name  wholesale_price  retail_price  sales  \
0          23        computer            500.0          1000    100   
1          96  Python Workout             35.0            75   1000   
2          97  Pandas Workout             35.0            75    500   
3          15          banana              0.5             1    200   
4          87        sandwich              3.0             5    300   

   net_revenue_per_product  wholesale@30%discount  new_revenue_per_product  \
0                  50000.0                 350.00                  65000.0   
1                  40000.0                  24.50                  50500.0   
2                  20000.0                  24.50                  25250.0   
3                    100.0                   0.35                    130.0   
4                    600.0                   2.10                    870.0   

   15%_tax_amount  revenue_after_15_tax  20%_tax_amount  revenue_after_20_tax  \
0          7500.0               42500.0         10000.0               40000.0   
1          6000.0               34000.0          8000.0               32000.0   
2          3000.0               17000.0          4000.0               16000.0   
3            15.0                  85.0            20.0                  80.0   
4            90.0                 510.0           120.0                 480.0   

   25%_tax_amount  revenue_after_25_tax  
0         12500.0               37500.0  
1         10000.0               30000.0  
2          5000.0               15000.0  
3            25.0                  75.0  
4           150.0                 450.0

In [111]:
# FIND THE AMOUNT OF TAX 15% OF REVENUE PER PRODUCT, ADD A NEW COLUMN 
# step 1 - find the amount of tax imposed 
# step 2 - Subtract the net_revenue_per_product - 15%_tax_amount
# step 3 - add the column 
# step 4 - get the difference before and after tax 

df['15%_tax_amount_after_discount'] = df['new_revenue_per_product'] * 0.15

df 

product_id            name  wholesale_price  retail_price  sales  \
0          23        computer            500.0          1000    100   
1          96  Python Workout             35.0            75   1000   
2          97  Pandas Workout             35.0            75    500   
3          15          banana              0.5             1    200   
4          87        sandwich              3.0             5    300   

   net_revenue_per_product  wholesale@30%discount  new_revenue_per_product  \
0                  50000.0                 350.00                  65000.0   
1                  40000.0                  24.50                  50500.0   
2                  20000.0                  24.50                  25250.0   
3                    100.0                   0.35                    130.0   
4                    600.0                   2.10                    870.0   

   15%_tax_amount  revenue_after_15_tax  20%_tax_amount  revenue_after_20_tax  \
0          7500.0               42500.0         10000.0               40000.0   
1          6000.0               34000.0          8000.0               32000.0   
2          3000.0               17000.0          4000.0               16000.0   
3            15.0                  85.0            20.0                  80.0   
4            90.0                 510.0           120.0                 480.0   

   25%_tax_amount  revenue_after_25_tax  15%_tax_amount_after_discount  
0         12500.0               37500.0                         9750.0  
1         10000.0               30000.0                         7575.0  
2          5000.0               15000.0                         3787.5  
3            25.0                  75.0                           19.5  
4           150.0                 450.0                          130.5

In [112]:
#step 2 - Subtract the net_revenue_per_product - 15%_tax_amount

df['revenue_after_15_tax_after_discount'] = df['new_revenue_per_product'] - df['15%_tax_amount_after_discount']

df 

product_id            name  wholesale_price  retail_price  sales  \
0          23        computer            500.0          1000    100   
1          96  Python Workout             35.0            75   1000   
2          97  Pandas Workout             35.0            75    500   
3          15          banana              0.5             1    200   
4          87        sandwich              3.0             5    300   

   net_revenue_per_product  wholesale@30%discount  new_revenue_per_product  \
0                  50000.0                 350.00                  65000.0   
1                  40000.0                  24.50                  50500.0   
2                  20000.0                  24.50                  25250.0   
3                    100.0                   0.35                    130.0   
4                    600.0                   2.10                    870.0   

   15%_tax_amount  revenue_after_15_tax  20%_tax_amount  revenue_after_20_tax  \
0          7500.0               42500.0         10000.0               40000.0   
1          6000.0               34000.0          8000.0               32000.0   
2          3000.0               17000.0          4000.0               16000.0   
3            15.0                  85.0            20.0                  80.0   
4            90.0                 510.0           120.0                 480.0   

   25%_tax_amount  revenue_after_25_tax  15%_tax_amount_after_discount  \
0         12500.0               37500.0                         9750.0   
1         10000.0               30000.0                         7575.0   
2          5000.0               15000.0                         3787.5   
3            25.0                  75.0                           19.5   
4           150.0                 450.0                          130.5   

   revenue_after_15_tax_after_discount  
0                              55250.0  
1                              42925.0  
2                              21462.5  
3                                110.5  
4                                739.5

In [113]:
#step 2 - Subtract the net_revenue_per_product - 15%_tax_amount

df['revenue_after_15_tax_after_discount'] = df['new_revenue_per_product'] - df['15%_tax_amount_after_discount']

df 

product_id            name  wholesale_price  retail_price  sales  \
0          23        computer            500.0          1000    100   
1          96  Python Workout             35.0            75   1000   
2          97  Pandas Workout             35.0            75    500   
3          15          banana              0.5             1    200   
4          87        sandwich              3.0             5    300   

   net_revenue_per_product  wholesale@30%discount  new_revenue_per_product  \
0                  50000.0                 350.00                  65000.0   
1                  40000.0                  24.50                  50500.0   
2                  20000.0                  24.50                  25250.0   
3                    100.0                   0.35                    130.0   
4                    600.0                   2.10                    870.0   

   15%_tax_amount  revenue_after_15_tax  20%_tax_amount  revenue_after_20_tax  \
0          7500.0               42500.0         10000.0               40000.0   
1          6000.0               34000.0          8000.0               32000.0   
2          3000.0               17000.0          4000.0               16000.0   
3            15.0                  85.0            20.0                  80.0   
4            90.0                 510.0           120.0                 480.0   

   25%_tax_amount  revenue_after_25_tax  15%_tax_amount_after_discount  \
0         12500.0               37500.0                         9750.0   
1         10000.0               30000.0                         7575.0   
2          5000.0               15000.0                         3787.5   
3            25.0                  75.0                           19.5   
4           150.0                 450.0                          130.5   

   revenue_after_15_tax_after_discount  
0                              55250.0  
1                              42925.0  
2                              21462.5  
3                                110.5  
4                                739.5

In [114]:
# step 3 - add the column 
#add the different columns then get the difference 

df.revenue_after_15_tax_after_discount.sum()

df.new_revenue_per_product.sum()

print('Revenue after 15% tax, with Discount is ', df.revenue_after_15_tax_after_discount.sum() , '$ ' ,' Revenue before 15% tax, with Discount is ' , df.new_revenue_per_product.sum(), '$' )

Revenue after 15% tax, with Discount is  120487.5 $   Revenue before 15% tax, with Discount is  141750.0 $


In [126]:
difference4 =  df.new_revenue_per_product.sum() - df.revenue_after_15_tax_after_discount.sum() 

difference4

21262.5

In [118]:
#confirmation 
141750.0 - 120487


21263.0

### FOR 15% SALES TAX WITHOUT DISCOUNT, THE STORE NETS 21263 LESS 

# WITH 20 % DISCOUNT 

In [120]:
df

product_id            name  wholesale_price  retail_price  sales  \
0          23        computer            500.0          1000    100   
1          96  Python Workout             35.0            75   1000   
2          97  Pandas Workout             35.0            75    500   
3          15          banana              0.5             1    200   
4          87        sandwich              3.0             5    300   

   net_revenue_per_product  wholesale@30%discount  new_revenue_per_product  \
0                  50000.0                 350.00                  65000.0   
1                  40000.0                  24.50                  50500.0   
2                  20000.0                  24.50                  25250.0   
3                    100.0                   0.35                    130.0   
4                    600.0                   2.10                    870.0   

   15%_tax_amount  revenue_after_15_tax  20%_tax_amount  revenue_after_20_tax  \
0          7500.0               42500.0         10000.0               40000.0   
1          6000.0               34000.0          8000.0               32000.0   
2          3000.0               17000.0          4000.0               16000.0   
3            15.0                  85.0            20.0                  80.0   
4            90.0                 510.0           120.0                 480.0   

   25%_tax_amount  revenue_after_25_tax  15%_tax_amount_after_discount  \
0         12500.0               37500.0                         9750.0   
1         10000.0               30000.0                         7575.0   
2          5000.0               15000.0                         3787.5   
3            25.0                  75.0                           19.5   
4           150.0                 450.0                          130.5   

   revenue_after_15_tax_after_discount  
0                              55250.0  
1                              42925.0  
2                              21462.5  
3                                110.5  
4                                739.5

In [121]:
# FIND THE AMOUNT OF TAX 15% OF REVENUE PER PRODUCT, ADD A NEW COLUMN 
# step 1 - find the amount of tax imposed 
# step 2 - Subtract the net_revenue_per_product - 20%_tax_amount
# step 3 - add the column 
# step 4 - get the difference before and after tax 

df['20%_tax_amount_after_discount'] = df['new_revenue_per_product'] * 0.2

df 

product_id            name  wholesale_price  retail_price  sales  \
0          23        computer            500.0          1000    100   
1          96  Python Workout             35.0            75   1000   
2          97  Pandas Workout             35.0            75    500   
3          15          banana              0.5             1    200   
4          87        sandwich              3.0             5    300   

   net_revenue_per_product  wholesale@30%discount  new_revenue_per_product  \
0                  50000.0                 350.00                  65000.0   
1                  40000.0                  24.50                  50500.0   
2                  20000.0                  24.50                  25250.0   
3                    100.0                   0.35                    130.0   
4                    600.0                   2.10                    870.0   

   15%_tax_amount  revenue_after_15_tax  20%_tax_amount  revenue_after_20_tax  \
0          7500.0               42500.0         10000.0               40000.0   
1          6000.0               34000.0          8000.0               32000.0   
2          3000.0               17000.0          4000.0               16000.0   
3            15.0                  85.0            20.0                  80.0   
4            90.0                 510.0           120.0                 480.0   

   25%_tax_amount  revenue_after_25_tax  15%_tax_amount_after_discount  \
0         12500.0               37500.0                         9750.0   
1         10000.0               30000.0                         7575.0   
2          5000.0               15000.0                         3787.5   
3            25.0                  75.0                           19.5   
4           150.0                 450.0                          130.5   

   revenue_after_15_tax_after_discount  20%_tax_amount_after_discount  
0                              55250.0                        13000.0  
1                              42925.0                        10100.0  
2                              21462.5                         5050.0  
3                                110.5                           26.0  
4                                739.5                          174.0

In [128]:
#step 2 - Subtract the net_revenue_per_product - 20%_tax_amount

df['revenue_after_20_tax_after_discount'] = df['new_revenue_per_product'] - df['20%_tax_amount_after_discount']

df 

product_id            name  wholesale_price  retail_price  sales  \
0          23        computer            500.0          1000    100   
1          96  Python Workout             35.0            75   1000   
2          97  Pandas Workout             35.0            75    500   
3          15          banana              0.5             1    200   
4          87        sandwich              3.0             5    300   

   net_revenue_per_product  wholesale@30%discount  new_revenue_per_product  \
0                  50000.0                 350.00                  65000.0   
1                  40000.0                  24.50                  50500.0   
2                  20000.0                  24.50                  25250.0   
3                    100.0                   0.35                    130.0   
4                    600.0                   2.10                    870.0   

   15%_tax_amount  revenue_after_15_tax  20%_tax_amount  revenue_after_20_tax  \
0          7500.0               42500.0         10000.0               40000.0   
1          6000.0               34000.0          8000.0               32000.0   
2          3000.0               17000.0          4000.0               16000.0   
3            15.0                  85.0            20.0                  80.0   
4            90.0                 510.0           120.0                 480.0   

   25%_tax_amount  revenue_after_25_tax  15%_tax_amount_after_discount  \
0         12500.0               37500.0                         9750.0   
1         10000.0               30000.0                         7575.0   
2          5000.0               15000.0                         3787.5   
3            25.0                  75.0                           19.5   
4           150.0                 450.0                          130.5   

   revenue_after_15_tax_after_discount  20%_tax_amount_after_discount  \
0                              55250.0                        13000.0   
1                              42925.0                        10100.0   
2                              21462.5                         5050.0   
3                                110.5                           26.0   
4                                739.5                          174.0   

   revenue_after_20_tax_after_discount  
0                              52000.0  
1                              40400.0  
2                              20200.0  
3                                104.0  
4                                696.0

In [123]:
# step 3 - add the column 
#add the different columns then get the difference 

df.revenue_after_20_tax_after_discount.sum()

df.new_revenue_per_product.sum()

print('Revenue after 20% tax, with Discount is ', df.revenue_after_20_tax_after_discount.sum() , '$ ' ,' Revenue before 20% tax, with Discount is ' , df.new_revenue_per_product.sum(), '$' )

Revenue after 20% tax, with Discount is  113400.0 $   Revenue before 20% tax, with Discount is  141750.0 $


In [124]:
difference5 =  df.new_revenue_per_product.sum() - df.revenue_after_20_tax_after_discount.sum() 

difference5

28350.0

In [125]:
#confirmation 
141750 - 113400

28350

### FOR 20% SALES TAX WITHOUT DISCOUNT, THE STORE NETS 228350 LESS

# WITH 25 % DISCOUNT 

In [130]:
# FIND THE AMOUNT OF TAX 25% OF REVENUE PER PRODUCT, ADD A NEW COLUMN 
# step 1 - find the amount of tax imposed 
# step 2 - Subtract the net_revenue_per_product - 25%_tax_amount
# step 3 - add the column 
# step 4 - get the difference before and after tax 

df['25%_tax_amount_after_discount'] = df['new_revenue_per_product'] * 0.25
df

product_id            name  wholesale_price  retail_price  sales  \
0          23        computer            500.0          1000    100   
1          96  Python Workout             35.0            75   1000   
2          97  Pandas Workout             35.0            75    500   
3          15          banana              0.5             1    200   
4          87        sandwich              3.0             5    300   

   net_revenue_per_product  wholesale@30%discount  new_revenue_per_product  \
0                  50000.0                 350.00                  65000.0   
1                  40000.0                  24.50                  50500.0   
2                  20000.0                  24.50                  25250.0   
3                    100.0                   0.35                    130.0   
4                    600.0                   2.10                    870.0   

   15%_tax_amount  revenue_after_15_tax  20%_tax_amount  revenue_after_20_tax  \
0          7500.0               42500.0         10000.0               40000.0   
1          6000.0               34000.0          8000.0               32000.0   
2          3000.0               17000.0          4000.0               16000.0   
3            15.0                  85.0            20.0                  80.0   
4            90.0                 510.0           120.0                 480.0   

   25%_tax_amount  revenue_after_25_tax  15%_tax_amount_after_discount  \
0         12500.0               37500.0                         9750.0   
1         10000.0               30000.0                         7575.0   
2          5000.0               15000.0                         3787.5   
3            25.0                  75.0                           19.5   
4           150.0                 450.0                          130.5   

   revenue_after_15_tax_after_discount  20%_tax_amount_after_discount  \
0                              55250.0                        13000.0   
1                              42925.0                        10100.0   
2                              21462.5                         5050.0   
3                                110.5                           26.0   
4                                739.5                          174.0   

   revenue_after_20_tax_after_discount  25%_tax_amount_after_discount  
0                              52000.0                        16250.0  
1                              40400.0                        12625.0  
2                              20200.0                         6312.5  
3                                104.0                           32.5  
4                                696.0                          217.5

In [131]:
#step 2 - Subtract the net_revenue_per_product - 25%_tax_amount

df['revenue_after_25_tax_after_discount'] = df['new_revenue_per_product'] - df['25%_tax_amount_after_discount']

df 

product_id            name  wholesale_price  retail_price  sales  \
0          23        computer            500.0          1000    100   
1          96  Python Workout             35.0            75   1000   
2          97  Pandas Workout             35.0            75    500   
3          15          banana              0.5             1    200   
4          87        sandwich              3.0             5    300   

   net_revenue_per_product  wholesale@30%discount  new_revenue_per_product  \
0                  50000.0                 350.00                  65000.0   
1                  40000.0                  24.50                  50500.0   
2                  20000.0                  24.50                  25250.0   
3                    100.0                   0.35                    130.0   
4                    600.0                   2.10                    870.0   

   15%_tax_amount  revenue_after_15_tax  20%_tax_amount  revenue_after_20_tax  \
0          7500.0               42500.0         10000.0               40000.0   
1          6000.0               34000.0          8000.0               32000.0   
2          3000.0               17000.0          4000.0               16000.0   
3            15.0                  85.0            20.0                  80.0   
4            90.0                 510.0           120.0                 480.0   

   25%_tax_amount  revenue_after_25_tax  15%_tax_amount_after_discount  \
0         12500.0               37500.0                         9750.0   
1         10000.0               30000.0                         7575.0   
2          5000.0               15000.0                         3787.5   
3            25.0                  75.0                           19.5   
4           150.0                 450.0                          130.5   

   revenue_after_15_tax_after_discount  20%_tax_amount_after_discount  \
0                              55250.0                        13000.0   
1                              42925.0                        10100.0   
2                              21462.5                         5050.0   
3                                110.5                           26.0   
4                                739.5                          174.0   

   revenue_after_20_tax_after_discount  25%_tax_amount_after_discount  \
0                              52000.0                        16250.0   
1                              40400.0                        12625.0   
2                              20200.0                         6312.5   
3                                104.0                           32.5   
4                                696.0                          217.5   

   revenue_after_25_tax_after_discount  
0                              48750.0  
1                              37875.0  
2                              18937.5  
3                                 97.5  
4                                652.5

In [132]:
# step 3 - add the column 
#add the different columns then get the difference 

df.revenue_after_25_tax_after_discount.sum()

df.new_revenue_per_product.sum()

print('Revenue after 25% tax, with Discount is ', df.revenue_after_25_tax_after_discount.sum() , '$ ' ,' Revenue before 25% tax, with Discount is ' , df.new_revenue_per_product.sum(), '$' )

Revenue after 25% tax, with Discount is  106312.5 $   Revenue before 25% tax, with Discount is  141750.0 $


In [133]:
difference6 =  df.new_revenue_per_product.sum() - df.revenue_after_25_tax_after_discount.sum() 

difference6

35437.5

In [134]:
#confirmation 
141750.0-106312.5 

35437.5

### FOR 25% SALES TAX WITHOUT DISCOUNT, THE STORE NETS 35437.5 LESS

In [139]:
lost_after_15_discount_per_product_after_discount = (df.new_revenue_per_product.sum()-df.revenue_after_15_tax_after_discount.sum())
lost_after_20_discount_per_product_after_discount = (df.new_revenue_per_product.sum()-df.revenue_after_20_tax_after_discount.sum())
lost_after_25_discount_per_product_after_discount = (df.new_revenue_per_product.sum()-df.revenue_after_25_tax_after_discount.sum())

In [142]:
print('loss after 15 % per_product after discount KSHS', lost_after_15_discount_per_product_after_discount)

print('loss after 20 % per product after discount KSHS', lost_after_20_discount_per_product_after_discount)

print('loss after 25 % per product after discount KSHS' , lost_after_25_discount_per_product_after_discount)

loss after 15 % per_product after discount KSHS 21262.5
loss after 20 % per product after discount KSHS 28350.0
loss after 25 % per product after discount KSHS 35437.5


In [143]:
print('loss after 15 % per_product KSHS', lost_after_15_discount_per_product)

print('loss after 20 % per product KSHS', lost_after_20_discount_per_product)

print('loss after 25 % per product KSHS' , lost_after_25_discount_per_product)

loss after 15 % per_product KSHS 21262.5
loss after 20 % per product KSHS 28350.0
loss after 25 % per product KSHS 35437.5


In [148]:
## PUT RESULTS IN A DF2 

print('loss after 15 % per_product after discount KSHS', lost_after_15_discount_per_product_after_discount)

print('loss after 20 % per product after discount KSHS', lost_after_20_discount_per_product_after_discount)

print ('loss after 25 % per product after discount KSHS' , lost_after_25_discount_per_product_after_discount)

print('loss after 15 % per_product before KSHS', lost_after_15_discount_per_product)

print ('loss after 20 % per product before KSHS', lost_after_20_discount_per_product)

print ('loss after 25 % per product before  KSHS' , lost_after_25_discount_per_product)

loss after 15 % per_product after discount KSHS 21262.5
loss after 20 % per product after discount KSHS 28350.0
loss after 25 % per product after discount KSHS 35437.5
loss after 15 % per_product before KSHS 21262.5
loss after 20 % per product before KSHS 28350.0
loss after 25 % per product before  KSHS 35437.5


In [149]:
#Convert into an np array , 


a = np.array([['loss after 15 % per_product after discount KSHS', 'loss after 20 % per product after discount KSHS','loss after 25 % per product after discount KSHS','loss after 15 % per_product KSHS','loss after 20 % per product KSHS',  'loss after 25 % per product KSHS' ],[ lost_after_15_discount_per_product_after_discount,lost_after_20_discount_per_product_after_discount, lost_after_25_discount_per_product_after_discount, lost_after_15_discount_per_product, lost_after_20_discount_per_product,  lost_after_25_discount_per_product ]])

In [152]:
#CONVERT TO DATAFRAME 
df2 = pd.DataFrame(a)

df2

0  \
0  loss after 15 % per_product after discount KSHS   
1                                          21262.5   

                                                 1  \
0  loss after 20 % per product after discount KSHS   
1                                          28350.0   

                                                 2  \
0  loss after 25 % per product after discount KSHS   
1                                          35437.5   

                                  3                                 4  \
0  loss after 15 % per_product KSHS  loss after 20 % per product KSHS   
1                           21262.5                           28350.0   

                                  5  
0  loss after 25 % per product KSHS  
1                           35437.5

In [153]:
df2

0  \
0  loss after 15 % per_product after discount KSHS   
1                                          21262.5   

                                                 1  \
0  loss after 20 % per product after discount KSHS   
1                                          28350.0   

                                                 2  \
0  loss after 25 % per product after discount KSHS   
1                                          35437.5   

                                  3                                 4  \
0  loss after 15 % per_product KSHS  loss after 20 % per product KSHS   
1                           21262.5                           28350.0   

                                  5  
0  loss after 25 % per product KSHS  
1                           35437.5

In [156]:
#find out shape then reshape 

df2.shape

(2, 6)

## GOD BLESS YOU AZUBI AFRICA FOR THE OPPORTUNITY 